src/main/kotlin/com/karumi/maxibonkata/Developer.kt

```diff
@@ -0,0 +1,14 @@
+package com.karumi.maxibonkata
+
+data class Developer internal constructor(val name: String = "", private val numberOfMaxibons: Int = 0) {
+    val maxibonsToGrab: Int = Math.max(0, numberOfMaxibons)
+
+    companion object {
+        val pedro = Developer("Pedro", 3)
+        val fran = Developer("Fran", 1)
+        val davide = Developer("Davide", 0)
+        val sergio = Developer("Sergio", 2)
+        val jorge = Developer("Jorge", 1)
+    }
+}
```

Comment:
This is the best way I've found to restrict the developers construction and provide just a bunch of developers available from outside this project. Remember that during our trainings we use this class as an example of how to hide the construction of a class but leave the type public. I know this could be implemented using a factory method, but I prefer this implementation.

Reply:
You could put them out of the class but on the same file. Then you don't need a `companion object` declaration.
The constructor will continue private but you have access to the different `Developers`

Reply:
I think this is the more kotlin-friendly way to do it (and it's what I'd have done) 👌 

I'd just add the named argument `numberOfMaxibons` when building all the developers so that it's easy to see what those `3`, `1` and `0` values are.

Reply:
As a way of keeping the namespace simulating a factory method, I prefer to keep the ``Developer`` companion object.

When talking about the named params, I've updated the karumies declaration to use ``numberOfMaxibons`` as a named param in construction.